src/ray/core_worker/task_manager.cc

```diff
@@ -972,8 +972,7 @@ bool TaskManager::RetryTaskIfPossible(const TaskID &task_id,
   // If the actor isn't dead and it's a user exception, we should update the seq no. If an
   // actor is dead and restarted, the seqno is reset, and we don't need to update it when
   // resubmitting a task.
-  bool update_seqno = error_info.error_type() != rpc::ErrorType::ACTOR_DIED &&
-                      error_info.error_type() != rpc::ErrorType::ACTOR_UNAVAILABLE;
+  bool update_seqno = error_info.error_type() != rpc::ErrorType::ACTOR_DIED;
```

Comment:
Why do we still keep this case instead of always updating seq_no?

Reply:
I think both should be fine. If an actor died, the new actor should not reject the tasks with the same `seq_no`.

Reply:
In the case of `ACTOR_DIED`, it would maintain the original order of execution in some cases (if multiple tasks need to get resubmitted)

Very very minor edge case though -- I would be fine with dropping this as well to unify the failure handling.